### PR TITLE
added a command to get a deploy group's latest deploy sha

### LINKED ISSUE #359: Switching from branches to tags made it harder to get currently deployed git sha
@asottile was using branches for a couple things, now that branches are gone and tags don't have a specific format (they have a date stuck in the middle) there's no good way to get the currently deployed git sha for a deploy group.

### PR DESCRIPTION
**Pull request by mjksmith**
@asottile 
Fixes internal ticket PAASTA-3591
Fixes #359

### CODE REVIEW

File: paasta_tools/cli/cmds/get_latest_deployment.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+from paasta_tools.cli.utils import validate_service_name
+from paasta_tools.generate_deployments_for_service import get_latest_deployment_tag
+from paasta_tools.remote_git import list_remote_refs
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import get_git_url
+
+
+def add_subparser(subparsers):
+    list_parser = subparsers.add_parser(
+        'get_latest_deployment',
```
Comment (nhandler) [inline paasta_tools/cli/cmds/get_latest_deployment.py]: Can we make this `get-latest-deployment` to be consistent with our other commands?

**Comment by nhandler** [comment]
I just tested this out, and it worked fine (confirmed that running something like `git show $(paasta get_latest_deployment)` works as expected. The one thing I noticed was that the script outputs a blank line before the SHA.

**Comment by mjksmith** [comment]
@nhandler I don't notice that extra line in testing

```
matts@dev2-devc:~/example_service (master) $ git show $(python ../paasta/paasta_tools/cli/cli.py get-latest-deployment -s example_service -i dev-stage.everything)
commit 3eaa22dff8a4963716c4903ff02054917d22a460
Author: Matthew Smith <matts@yelp.com>
Date:   Tue Feb 2 11:30:14 2016 -0800

    noop
```

**Comment by nhandler** [comment]
Yeah, I'm not able to reproduce anymore either. And your code looks sane. Might have just been my shell doing something funky.

Ship It!

**Comment by Rob-Johnson** [comment]
shipit!

**Pull request merged**